public/app/plugins/datasource/azuremonitor/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -102,7 +109,7 @@ export const AzureCredentialsForm = (props: Props) => {
       {credentials.authType === 'clientsecret' && (
         <AppRegistrationCredentials
           credentials={credentials}
-          azureCloudOptions={legacyAzureCloudOptions}
+          azureCloudOptions={getAzureCloudOptions()}
```

Comment:
This change won't work as expected due to the differences between legacy and current cloud naming. It could lead to the UI being invalid for data sources that were created before the switch.

Reply:
The reason I thought this would be OK was because the values are always normalized on the backend where they seem to be consumed.  What have I missed on the UI that actually depends on these legacy names?  The only thing I can think of is maybe there is some odd behavior for existing self-hosted grafana servers configured with the cloud name using these legacy values?  If so, maybe we could change this to normalize the configured cloud to the newer names instead of holding onto the old names?

Reply:
Correct, existing data sources configured with a legacy cloud name would be unable to correctly display the cloud name if we remove this code. I think a normalisation function sounds reasonable.